docs/design_proposals/new-render-and-deploy.md

```diff
@@ -0,0 +1,374 @@
+      - image: <render.validators.customize[0].name> 
+      # specific arguments that could defer per whitelisted tool.
+        configPath/configPath: 
+
+      - image: <render.validators.customize[1].name> 
+      ...
+
+``` 
+
+The following graph describes the commands and operations to generate the kpt sources.
+
+![builtin render](images/builtin-render.png)
+
+### Integration of `kpt` live  and `skaffold` deploy
+
+`kpt` relies on `ResourceGroup` to accurately deploy the WET configurations to the cluster.  This requires the ResourceGroup CRD to be installed on the cluster side. 
+
+`skaffold dev` and `skaffold deploy` commands will detect and ask for user permissions before installing the `ResourceGroup` controller. It calls the following `kpt` commands. 
+
+![deploy command](images/deploy-cmd.png)
+
+*Note*: `skaffold` v2 will not provide the migration from `kpt` v0.X to `kpt` v1. Users can follow [this guide](https://googlecontainertools.github.io/kpt/reference/live/migrate/) to establish the `kpt` migration themselves.
+
+### Builtin `kpt`
+
+`skaffold` will have `kpt` embedded via go bindata, which ship together with the skaffold binary to guarantee the versions match.
+
+Currently, `skaffold` requires users to install `helm`, `kubectl`, `kustomize` and `kpt` and manage the version skew themselves. By making `kpt` and `skaffold` a bundle, users no longer need to install ANY other tools (`helm` and `kustomize` are used as `kpt` functions).
+
+## Migration Plan
+
+### `Helm` "deployer" 
+
+Customers can choose either to continue using `helm` deployer or migrate to `kpt` in both "render" and "deploy" stages. We provide users an easy approach to migrate `helm` "deployer" to the new skaffold. But if users continue using `helm`, we will not provide no further support for the new `kpt` features.
+
+This guarantees the `skaffold` v2 is backward compatible with `helm` and users have a seamless experience when introduced to the new render UI. 
+
+#### The migration command
+
+Users can run the existing skaffold `fix` command to migrate to the new skaffold.
+
+This will wipe and recreate the `skaffold.yaml` as below. After the migration, if users make a `helm` config change in `skaffold.yaml`, skaffold dev (and other skaffold commands enabling file watcher) will recreate the following kpt resources.  
+
+![Helm migration](images/helm-migration.png)
+
+#### Under the hood
+
+Though the new UI looks similar except moving helm from "deploy" to "render", the actual implementation is very different. Instead of calling the `helm` CLI, the `skaffold-helm-inflator` functions are used to convert the `Helm` charts to a kpt [`ResourceList.items`]（https://googlecontainertools.github.io/kpt/guides/producer/functions/#resourcelistitems). This function is expected to digest all the `helm` arguments listed in the `skaffold.yaml` reference doc [v2beta16](https://skaffold.dev/docs/references/yaml/#deploy-helm).
```

Comment:
Are there any features/flags that skaffold+helm supports currently that are not backwards compatible with the skaffold-helm-inflator?  If so they might be worth calling out in the migration story.

Reply:
Good point. The helm-inflator image is not there yet, we will add more doc once that part of work is more stabilized.